### PR TITLE
Fixed #21429 -- Added builtin logging support to BaseCommand.

### PR DESCRIPTION
**Pull request by berkerpeksag**
https://code.djangoproject.com/ticket/21429

### CODE REVIEW

**Comment by berkerpeksag** [comment]
Done.

* assertLogs commit: https://github.com/berkerpeksag/django/commit/1b4f10963628bb68246c193f1124a1f7b6e4c696
* patch_logger -> assertLogs commit: https://github.com/berkerpeksag/django/commit/6d534e40a1947da1eefc9e72278534dbadc6b901

File: docs/topics/logging.txt [hunk 1]
```diff
@@ -650,9 +657,17 @@ logging module.
...
+
+* The ``django.request`` logger is configured to send all messages with
+  ``ERROR`` or ``CRITICAL`` level to :class:`AdminEmailHandler`, as long as the
+  :setting:`DEBUG` setting is set to ``False``.
+
+* The ``django.security`` logger is configured the same as the
+  ``django.request`` logger, and any error events will be mailed to admins.
+
+* The ``django.commands`` logger is configured to send all messages with
+  ``INFO`` level to ``stdout``, ``WARNING`` and ``ERROR`` level to ``stderr``.
```
Comment (claudep) [inline docs/topics/logging.txt]: That does not match the docs above which says that BaseCommand.warning outputs to stdout.

Comment (berkerpeksag) [inline docs/topics/logging.txt]: Good catch, thanks. BaseCommand.warning documentation was wrong.

File: django/contrib/gis/tests/geoapp/tests.py [hunk 2]
```diff
@@ -208,9 +209,14 @@ def test_dumpdata_loaddata_cycle(self):
         """
         Test a dumpdata/loaddata cycle with geographic data.
         """
+        orig_stdout = sys.stdout
```
Comment (timgraham) [inline django/contrib/gis/tests/geoapp/tests.py]: We have `django.test.utils.captured_stdout` now.

**Comment by timgraham** [comment]
Closing due to inactivity.

**Pull request closed without merging**